### PR TITLE
Add support for elaborated types.

### LINKED ISSUE #1: current bindgen is broken for rust-mozjs
It reports an error about the copyability of inlineStorageLatin1, but the real problem appears to be that while it translated the Latin1Char type correctly, inside the unions it just has them as c_void.

### PR DESCRIPTION
**Pull request by metajack**
Things like JS::Latin1Char now show up as elaborated types and we need to use clang_Type_getNamedType to resolve them.

This should fix #1.

### CODE REVIEW

**Comment by metajack** [comment]
cc @emilio @nox

**Comment by metajack** [comment]
@edunham This repo is not reported to IRC or homu yet. Can you turn that on?

**Comment by nox** [comment]
So that's something that changed in libclang itself or something like that?

**Comment by emilio** [comment]
@nox It seems so.

@metajack Thanks for digging into this, my laptop OOMd the first time I tried to build LLVM 3.9 so I left it building again with `-j1` and went to bed. Adding a test for this would be ideal (`tests/headers/x.hpp` + `make test`, yeah, I know), but I can take care of it.

File: src/clang.rs [hunk 1]
```diff
@@ -946,6 +952,9 @@ pub fn type_to_str(x: Enum_CXTypeKind) -> &'static str {
         CXType_IncompleteArray => "IncompleteArray",
         CXType_VariableArray => "VariableArray",
         CXType_DependentSizedArray => "DependentSizedArray",
+        CXType_MemberPointer => "CXType_MemberPointer",
+        CXType_Auto => "CXType_Auto",
+        CXType_Elaborated => "CXType_Elaborated",
```
Comment (nox) [inline src/clang.rs]: Remove the `CXType_` prefixes.

Comment (metajack) [inline src/clang.rs]: Fixed.

**Comment by metajack** [comment]
@emilio Tests added and @nox's comment addressed.

**Comment by metajack** [comment]
@emilio re-r?

**Comment by emilio** [comment]
r+, I'm looking into CI issues right now (#5) that should allow us to autoland this instead of merging manually, but feel free to do so.

**Pull request closed without merging**